src/algorithms/reco/TrackClusterMatch.cc

```diff
@@ -0,0 +1,95 @@
+
+            std::optional<edm4eic::TrackSegment> closest_segment;
+            int closest_segment_id = -1;
+            double closest_distance = m_cfg.matching_distance;
+            // Loop through each track segment, and its points
+            for (int closest_id = 0; auto track : *tracks) {
+                if (used_tracks.count(closest_id) > 0) {
+                    m_log->trace("Skipping track segment already used");
+                    continue;
+                }
+                for (auto point : track.getPoints()) {
+                    // Check if the point is at the calorimeter
+                    // int id = m_detector->volumeManager().lookupDetector(cluster.getHits()[0].getCellID()).id(); // TODO: Find programmatic way to get detector cluster is from
+                    uint32_t ecal_barrel_id = 101;
+                    uint32_t hcal_barrel_id = 111;
+                    bool is_ecal = point.system == ecal_barrel_id;
+                    bool is_hcal = point.system == hcal_barrel_id;
+                    bool is_surface = point.surface == 1;
+
+                    if (!(is_ecal || is_hcal) || !is_surface) {
+                        m_log->trace("Skipping track point not at the calorimeter");
+                        continue;
+                    }
+                    const double track_eta = edm4hep::utils::eta(point.position);
+                    const double track_phi = edm4hep::utils::angleAzimuthal(point.position);
+                    m_log->trace("Track point at eta={}, phi={}", track_eta, track_phi);
+
+                    double delta = distance(cluster.getPosition(), point.position);
+                    m_log->trace("Distance between cluster and track point: {}", delta);
+                    if (delta < closest_distance) {
+                        closest_distance = delta;
+                        closest_segment = track;
+                        closest_segment_id = closest_id;
+                    }
+                }
+                closest_id++;
+            }
+            // If we found a point, create a new particle
+            if (closest_segment) {
+                m_log->trace("Found a closest point at distance {}", closest_distance);
+                auto particle = matched_particles->create();
+                particle.setCluster(cluster);
+                particle.setTrack(closest_segment.value().getTrack());
+                used_tracks.insert(closest_segment_id);
+            }
+        }
+    }
+
+    double TrackClusterMatch::distance(const edm4hep::Vector3f& v1, const edm4hep::Vector3f& v2) const {
+        double cluster_eta = edm4hep::utils::eta(v1);
+        double cluster_phi = edm4hep::utils::angleAzimuthal(v1);
+        double track_eta = edm4hep::utils::eta(v2);
+        double track_phi = edm4hep::utils::angleAzimuthal(v2);
+        double deta = cluster_eta - track_eta;
+        double dphi = cluster_phi - track_phi;
```

Comment:
```c++
static double Phi_mpi_pi(double phi) {
      return std::remainder(phi, 2 * M_PI);
}
```
```suggestion
        double dphi = Phi_mpi_pi(cluster_phi - track_phi);
```

Reply:
To clarify, is `Phi_mpi_pi` defined in some common header I can include, or just add it to my  code?

Reply:
No common headers so far. Just add it in the .cc file.